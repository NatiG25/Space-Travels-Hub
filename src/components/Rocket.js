import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { cancelRocket, fetchRockets, reserveRocket } from './redux/rockets/rockets';
import RocketContainer from './Rockets';

const Rockets = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchRockets());
  }, []);
  const rocketInfo = useSelector((state) => state.rocketReducer);

  const reserveRocketHandler = (id) => {
    dispatch(reserveRocket(id));
  };

  const cancelRocketHandler = (id) => {
    dispatch(cancelRocket(id));
  };

  return (
    <>
      {rocketInfo.map((rocket) => (
        <RocketContainer
          key={rocket.id}
          name={rocket.rocket_name}
          description={rocket.description}
          image={rocket.flickr_images[0]}
          reserved={rocket.reserved}
          reserveRocketHandler={() => reserveRocketHandler(rocket.id)}
          cancelRocketHandler={() => cancelRocketHandler(rocket.id)}
        />
      ))}
    </>
  );
};
export default Rockets;
