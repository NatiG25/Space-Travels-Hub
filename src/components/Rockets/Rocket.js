import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { cancelRocket, fetchRockets, reserveRocket } from '../redux/rockets/rockets';
import RocketContainer from './Rockets';

const Rockets = () => {
  const rocketInfo = useSelector((state) => state.rocketReducer);

  const dispatch = useDispatch();
  useEffect(() => {
    if (!rocketInfo.length) dispatch(fetchRockets());
  }, []);

  const reserveRocketHandler = (id) => {
    dispatch(reserveRocket(id));
  };

  const cancelRocketHandler = (id) => {
    dispatch(cancelRocket(id));
  };

  return (
    <>
      {rocketInfo.map((rocket) => (
        <RocketContainer
          key={rocket.id}
          name={rocket.rocket_name}
          description={rocket.description}
          image={rocket.flickr_images[0]}
          reserved={rocket.reserved}
          reserveRocketHandler={() => reserveRocketHandler(rocket.id)}
          cancelRocketHandler={() => cancelRocketHandler(rocket.id)}
        />
      ))}
    </>
  );
};
export default Rockets;
