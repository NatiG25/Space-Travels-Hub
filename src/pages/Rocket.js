import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../api/fetchRocketData';
import RocketContainer from '../components/Rockets';

const Rockets = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, []);
  const rocketInfo = useSelector((state) => state.rocketInfo);
  return (
    <>
      {rocketInfo.map((rocket) => (
        <RocketContainer
          key={rocket.id}
          name={rocket.rocket_name}
          description={rocket.description}
          image={rocket.flickr_image}
        />
      ))}
    </>
  );
};

export default Rockets;
