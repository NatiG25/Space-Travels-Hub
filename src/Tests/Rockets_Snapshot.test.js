import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../components/redux/configureStore'
import Rocket from '../components/Rockets/Rocket';
import RocketContainer from '../components/Rockets/Rockets';

describe('Test all Rockets components', () => {
  it('renders Rocket correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Rocket /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Rocket correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><RocketContainer /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
