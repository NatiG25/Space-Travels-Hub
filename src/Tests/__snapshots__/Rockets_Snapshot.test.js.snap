// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test all Rockets components renders Rocket correctly 1`] = `null`;

exports[`Test all Rockets components renders Rocket correctly 2`] = `
<section
  className="rocket"
>
  <div>
    <img />
  </div>
  <div>
    <h2 />
    <p />
    <button
      className="book"
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</section>
`;
